src/server/tiered_storage.cc

```
@@ -68,17 +81,14 @@ class TieredStorageV2::ShardOpManager : public tiering::OpManager {
     return false;
   }
 
-  void ReportStashed(EntryId id, tiering::DiskSegment segment) override {
-    if (holds_alternative<string_view>(id)) {
-      SetExternal(get<string_view>(id), segment);
-    } else {
-      for (const auto& [sub_key, sub_segment] :
-           ts_->bins_->ReportStashed(get<tiering::SmallBins::BinId>(id), segment))
-        SetExternal(string_view{sub_key}, sub_segment);
-    }
+  void ReportStashed(EntryId id, tiering::DiskSegment segment, error_code ec) override {
+    if (ec)
```

we should report the io errors

---

you mean add them to stats? log them?

---

we can start with logging them